### ISSUE TITLE
BUG: Undefined behavior of Series.str.split

### ISSUE THREAD

**Issue by YarShev** [pandas-dev/pandas#51400]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas

pandas_df = pandas.DataFrame(["asd"])
print(pandas_df.dtypes)
# 0    object
# dtype: object

pandas_df2 = pandas.DataFrame([1])
print(pandas_df2.dtypes)
# 0    int64
# dtype: object

pandas_df3 = pandas.concat([pandas_df, pandas_df2], axis=0)
print(pandas_df3.dtypes)
# 0    object
# dtype: object

print(pandas_df3[0].str.split())
# 0    [asd]
# 0      NaN
# Name: 0, dtype: object

print(pandas_df[0].str.split())
# 0    [asd]
# Name: 0, dtype: object

print(pandas_df2.astype("string")[0].str.split())
# 0    [1]
# Name: 0, dtype: object

print(pandas_df2.astype("object")[0].str.split())
AttributeError: Can only use .str accessor with string values!
```

### Issue Description

It seems that there is some inconsistency when calling `str.split`. `pandas_df3` has `object` dtype and when calling `str.split` it returns `NaN` for `1` value. However, when we convert `pandas_df`, which has `int` dtype, to `string`, when calling `str.split` it returns `[1]` for `1` value. And also, when we convert `pandas_df`, which has `int` dtype, to `object` as `pandas_df` has, when calling `str.split` it raises the error. 

### Expected Behavior

It seems that only what `print(pandas_df2.astype("string")[0].str.split())` returns would be the right behavior.

### Installed Versions

<details>

pandas : 1.5.3
numpy : 1.24.1
pytz : 2022.7.1
dateutil : 2.8.2
setuptools : 66.1.1
pip : 22.3.1
Cython : None
pytest : 7.2.1
hypothesis : None
sphinx : 6.1.3
blosc : None
feather : 0.4.1
xlsxwriter : None
lxml.etree : 4.9.2
html5lib : None
pymysql : None
psycopg2 : 2.9.3
jinja2 : 3.1.2
IPython : 8.8.0
pandas_datareader: None
bs4 : 4.11.1
bottleneck : None
brotli :
fastparquet : 2022.12.0
fsspec : 2023.1.0
gcsfs : None
matplotlib : 3.6.3
numba : None
numexpr : 2.8.3
odfpy : None
openpyxl : 3.0.10
pandas_gbq : 0.15.0
pyarrow : 10.0.1
pyreadstat : None
pyxlsb : None
s3fs : 2023.1.0
scipy : 1.10.0
snappy : None
sqlalchemy : 1.4.46
tables : 3.7.0
tabulate : None
xarray : 2023.1.0
xlrd : 2.0.1
xlwt : None
zstandard : None
tzdata : None

</details>

**Comment by asishm** [comment]
It's a consequence of `pandas_df3` having an `object` column where one item is an int and the other a string. Pandas does not coerce the `1` to a string when doing the concat.

The main inconsistency is that (imo) the second example below raises

```python
In [17]: pd.Series([1, 'a'], dtype="object").str.split()
Out[17]:
0    NaN
1    [a]
dtype: object

In [18]: pd.Series([1], dtype="object").str.split() # raises
```

**Comment by YarShev** [comment]
Correct! Another option would also be to return NaN in your second example but I am not sure users expect such a behavior.

**Comment by rmhowe425** [comment]
take
